controllers/discordactions.js

```diff
@@ -63,6 +64,89 @@ const createGroupRole = async (req, res) => {
   }
 };
 
+/**
+ * Controller function to handle the soft deletion of a group role.
+ *
+ * @param {Object} req - The request object
+ * @param {Object} res - The response object
+ * @returns {Promise<void>}
+ */
+const deleteGroupRole = async (req, res) => {
+  const isDevMode = req.query.dev === "true";
+
+  if (isDevMode) {
+    try {
+      const { groupId } = req.params;
+
+      const getRoleId = async (groupId) => {
+        try {
+          const roleDoc = await admin.firestore().collection("discord-roles").doc(groupId).get();
+          return roleDoc.data().roleid;
+        } catch (error) {
+          logger.error("Error fetching roleId by groupId", error);
+          throw error;
+        }
+      };
+
+      const roleid = await getRoleId(groupId);
+
+      const { roleExists, existingRoles } = await discordRolesModel.isGroupRoleExists({ roleid });
+
+      if (!roleExists) {
+        return res.status(404).json({
+          error: "Group role not found",
+        });
+      }
+
+      const roleData = existingRoles.docs[0].data();
+
+      // To be implemented later after availability of Discord API
+
+      // const discordDeletionSuccess = await discordServices.deleteGroupRoleFromDiscord(roleid);
+
+      // if (!discordDeletionSuccess) {
+      //   return res.status(500).json({
+      //     error: "Failed to delete role from Discord server",
+      //   });
+      // }
```

Comment:
Why are we not implementing this as Pankaj has done with this?

Reply:
Initially, since the API was not implemented, Ankush divided this task into two parts and limited the scope of this PR to db changes only and directed the discord integration into a separate task to Pankaj.

And, due to some issues, this task has already been delayed too much that's why I was planning to go with the initial flow only, integrating in separate PR.

Suggestions are welcome.